src/com/lftechnology/java_training/jan7/FileInput.java

```diff
@@ -0,0 +1,153 @@
+package com.lftechnology.java_training.jan7;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.util.Scanner;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Write a program called FileInput to read an int, a double, and a String form a text file called "in.txt", and produce the following
+ * output: (Hint use Scanner to read from file)
+ * <ol>
+ * <li>The integer read is 12</li>
+ * <li>The floating point number read is 33.44</li>
+ * <li>The String read is"Peter"</li>
+ * <li>Hi! Peter, the sum of 12 and 33.44 is 45.44</li>
+ * </ol>
+ * 
+ * @author UtsabNeupane<utsabneupane@lftechnology.com>
+ * 
+ */
+public class FileInput {
+	private static int[] intReadValue = new int[10];
+	private static double[] doubleReadValue = new double[10];
+	private static String[] stringReadValue = new String[10];
+	private static int intValueCounter = 0;
+	private static int doubleValueCounter = 0;
+	private static int stringValueCounter = 0;
+	private static final Logger LOGGER = Logger.getLogger(FileInput.class.getName());
+	private static final String fileName = "in.txt";
+
+	public static void main(String[] args) {
+		// read values from file
+		boolean readStatus = readFromFile();
+		if (readStatus) {
+			// display values written in file to console
+			displayValueFromFile();
+		}
+	}
+
+	/**
+	 * read data from file and store it according to its type
+	 */
+	private static boolean readFromFile() {
+		Scanner inputFile = null;
+		try {
+			inputFile = new Scanner(new BufferedReader(new FileReader(fileName)));
+			// loop until the value finishes
+			while (inputFile.hasNext()) {
+				// check if value is int
+				if (inputFile.hasNextInt()) {
+					intReadValue[intValueCounter++] = Integer.parseInt(inputFile.next());
+					// check if value is double
+				} else if (inputFile.hasNextDouble()) {
+					doubleReadValue[doubleValueCounter++] = Double.parseDouble(inputFile.next());
+				} else {
+					stringReadValue[stringValueCounter++] = inputFile.next();
+
+				}
+			}
+			return true;
+		} catch (Exception e) {
+			LOGGER.log(Level.SEVERE, "file not found {0}", e);
+			return false;
+
+		} finally {
+			inputFile.close();
```

Comment:
Possible null pointer exception detected. When you initialize it as null out of try block and exception is encountered, it'll go to final block with null value.